delivery_carrier_seur/model/stock.py

```diff
@@ -0,0 +1,167 @@
+        selection='_get_seur_products', string='Seur Product Code',
+        default=False)
+
+    def _get_seur_services(self):
+        return self.env['delivery.carrier'].SEUR_SERVICES
+
+    def _get_seur_products(self):
+        return self.env['delivery.carrier'].SEUR_PRODUCTS
+
+    @api.onchange('carrier_id')
+    def carrier_id_onchange(self):
+        if not self.carrier_id:
+            return
+        carrier = self.carrier_id
+        self.seur_service_code = carrier.seur_service_code
+        self.seur_product_code = carrier.seur_product_code
+
+    @api.multi
+    def _generate_seur_label(self, package_ids=None):
+        self.ensure_one()
+        if not self.carrier_id.seur_config_id:
+            raise exceptions.Warning(_('No SEUR Config defined in carrier'))
+        if not self.picking_type_id.warehouse_id.partner_id:
+            raise exceptions.Warning(
+                _('Please define an address in the %s warehouse') % (
+                    self.warehouse_id.name))
+
+        config = self.carrier_id.seur_config_id
+
+        seur_context = {
+            'printer': 'ZEBRA',
+            'printer_model': 'LP2844-Z',
+            'ecb_code': '2C',
+        }
+
+        if config.file_type == 'pdf':
+            seur_context['pdf'] = True
+
+        seur_picking = Picking(
+            config.username,
+            config.password,
+            config.vat,
+            config.franchise_code,
+            'Odoo',  # seurid
+            config.integration_code,
+            config.accounting_code,
+            seur_context
+        )
+
+        try:
+            connect = seur_picking.test_connection()
+            if connect != 'Connection successfully':
+                raise exceptions.Warning(
+                    _('Error conecting with SEUR:\n%s' % connect))
+        except HTTPError, e:
+            raise exceptions.Warning(
+                _('Error conecting with SEUR try later:\n%s' % e))
+
+        data = self._get_label_data()
+        tracking_ref, label, error = seur_picking.create(data)
+
+        if error:
+            raise exceptions.Warning(
+                _('Error sending label to SEUR\n%s' % error))
+
+        self.write({'carrier_tracking_ref': tracking_ref})
+
+        return [{
+            'name': self.name + '_' + tracking_ref + '.' + config.file_type,
+            'file': label,
+            'file_type': config.file_type
+        }]
+
+    def _get_label_data(self):
+        partner = self.partner_id.parent_id or self.partner_id
+        data = {
+            'servicio': unidecode(self.seur_service_code),
+            'product': unidecode(self.seur_product_code),
+            'total_bultos': self.number_of_packages or '1',
+            'total_kilos': self.weight or '1',
+            'peso_bulto': self.weight_net or '1',
+            'observaciones': self.note or '',
+            'referencia_expedicion': unidecode(self.name),
+            'ref_bulto': '',
+            'clave_portes': '',  # F
```

Comment:
Corrected. I't was only a comment because I didn't know if use 'F' or '' but finally is always 'F'.